app/utils/data.py

```
@@ -0,0 +1,21 @@
+# 모듈 위치 기준으로 데이터 파일 경로 설정
+BASE_DIR = Path(__file__).resolve().parent.parent
+USERS_FILE = BASE_DIR / "data" / "users.json"
+
+def read_users():
+    try:
+        with open(USERS_FILE, encoding="utf-8") as f:
+            return json.load(f)
+    except FileNotFoundError:
+        return []
+    except json.JSONDecodeError as e:
+        raise ValueError(f"Invalid JSON format in users.json: {e}")
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)